### PR TITLE
Drop support for EOL Python 2.7

### PR DESCRIPTION
**Pull request by hugovk**
For https://github.com/robotframework/SeleniumLibrary/issues/1444.

* Drop support for EOL Python 2.7
* Upgrade Python syntax with https://github.com/asottile/pyupgrade `--py36-plus`
* Update Trove classifiers and python_requires
* Remove Jython from docs

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -10,9 +10,9 @@ SeleniumLibrary_ is a web testing library for `Robot Framework`_ that
 utilizes the Selenium_ tool internally. The project is hosted on GitHub_
 and downloads can be found from PyPI_.
 
-SeleniumLibrary works with Selenium 3 and 4. It supports Python 2.7 as well as
```
Comment (aaltat) [inline README.rst]: Could you this file as it is for while. It will be updated when 5.0 is release is made.

File: docs/index.html [hunk 2]
```diff
@@ -32,7 +32,7 @@ <h1><a class="toc-backref" href="#id6">Introduction</a></h1>
 and downloads can be found from <a class="reference external" href="https://pypi.python.org/pypi/robotframework-seleniumlibrary">PyPI</a>.</p>
 <p>SeleniumLibrary works with Selenium 3 and 4. It supports Python 2.7 as well as
 Python 3.6 or newer. In addition to the normal <a class="reference external" href="https://python.org">Python</a> interpreter, it
-works also with <a class="reference external" href="https://pypy.org">PyPy</a> and <a class="reference external" href="https://jython.org/">Jython</a>. Unfortunately <a class="reference external" href="https://www.seleniumhq.org/">Selenium</a> is not
```
Comment (aaltat) [inline docs/index.html]: This is updated in the release process, could you revert the changes.

**Review by aaltat — CHANGES_REQUESTED** [review]
There was few changes that are done only when 5.0 release is made. Could you revert those changes and then this is ready for merge.

**Comment by aaltat** [comment]
Thank you for the work. I had two change request, but other than those,  this looks good to be merged.

File: README.rst (hunk 1 above)
Comment (hugovk) [reply README.rst]: Reverted!

File: docs/index.html (hunk 2 above)
Comment (hugovk) [reply docs/index.html]: Reverted!

**Review by aaltat — APPROVED** [review]
